cmd/av/stack.go

```diff
@@ -83,7 +85,65 @@ stack. This is useful for rebasing a whole stack on the latest changes from the
 base branch.
 `),
 	RunE: func(cmd *cobra.Command, args []string) error {
-		return errors.New("unimplemented")
+		repo, err := getRepo()
+		if err != nil {
+			return err
+		}
+		originalBranch, err := repo.CurrentBranchName()
+		if err != nil {
+			return err
+		}
+
+		defer func() {
+			if _, err := repo.CheckoutBranch(&git.CheckoutBranch{Name: originalBranch}); err != nil {
+				logrus.WithError(err).Warnf("failed to reset to original branch: %q", originalBranch)
+			}
+		}()
+
+		root, err := stacks.GetCurrentRoot(repo)
+		if err != nil {
+			return err
+		}
+		printStackTree(root, 0)
+
+		if len(root.Next) == 0 {
+			// this shouldn't happen, but just in case
+			return errors.New("no branches to sync")
+		}
+
+		current := root.Next[0]
+		for {
+			if _, err := repo.CheckoutBranch(&git.CheckoutBranch{
+				Name: current.Branch.Name,
+			}); err != nil {
+				return errors.WrapIff(err, "failed to checkout branch %q", current.Branch.Name)
+			}
+			res, err := stacks.SyncBranch(repo, &stacks.SyncBranchOpts{
+				Parent: current.Branch.Parent,
+			})
+			if err != nil {
+				return errors.WrapIff(err, "failed to sync branch %q", current.Branch.Name)
+			}
+			switch res.Status {
+			case stacks.SyncAlreadyUpToDate:
+				fmt.Printf("Branch %q is already up-to-date with %q\n", current.Branch.Name, current.Branch.Parent)
+			case stacks.SyncUpdated:
+				fmt.Printf("Branch %q synchronized with %q\n", current.Branch.Name, current.Branch.Parent)
+			case stacks.SyncConflict:
```

Comment:
I'm assuming this is where we have a TODO?

Reply:
Yep. Working on this right now.